### ISSUE TITLE
CML Aromatic Bonds not being written

### ISSUE THREAD

**Issue by kmlefran** [openbabel/openbabel#2646]
## Environment Information
Open Babel version: 3.1.1 (also on  any 3.0.0 or later)
Operating system and version: Mac OS Ventura/CentOS Linux

## Expected Behavior
Given aromatic bond orders in .cml file bondArray, should write aromatic SMILES, e.g.  c1ccccc1 for benzene
Should also write bond type as "4" to .mol file

## Actual Behavior
Writes aromatic bonds as radicals, e.g. [CH]1[CH][CH][CH][CH][CH]1

Writes bond type as "5" to mol file, e.g.

[aromatic_mol.txt](https://github.com/openbabel/openbabel/files/13416854/aromatic_mol.txt)

## Steps to Reproduce
[aromatic_cml.txt](https://github.com/openbabel/openbabel/files/13416829/aromatic_cml.txt)

```
obabel -ismi -:"c1ccccc1" -ocml -xA -O aromatic_cml.txt

obabel -icml aromatic_cml.txt -osmi

obabel -icml aromatic_cml.txt -omol -O aromatic_mol.txt
```

Note that Open Babel 2.4.1 writes a SMILES that has aromaticity

`obabel -icml aromatic_cml.txt -osmi`

[cH]1[cH][cH][cH][cH][cH]1

**Comment by kmlefran** [comment]
> Try to replicate the following
> 
> ```shell
> $ obabel -:"c1ccccc1" -O aromatic.cml -A
> 1 molecule converted
> $ obabel aromatic.cml -ocan
> c1ccccc1	
> 1 molecule converted
> ```
> 
> as observed with OpenBabel (version 3.1.1 provided by Debian 13/trixie).

Thanks for the tips on the redundancies in the command.

I ran the commands as you said, and that works, but the aromatic.cml created does not write bond orders as aromatic, it writes the Kekule form. 
[aromatic.txt](https://github.com/openbabel/openbabel/files/13417932/aromatic.txt)

Changing the -A flag in your prompt to -xA writes the bonds as aromatic and I still see the issue described

```
$ obabel -:"c1ccccc1" -O aromatic.cml -xA
1 molecule converted
$ obabel aromatic.cml -ocan
[CH]1[CH][CH][CH][CH][CH]1
1 molecule converted
```

Also for the comment on version, we are currently using the most recent 3.1.1, but I noticed the error we see in 3.1.1 isn't seen in 2.4.1

**Comment by nbehrnd** [comment]
Ok, we now exclude the round trip .smi -> .cml -> .smi.  If I request the output back to the CLI to halt at level of the .cml, I observe two different results.

Either a benzene formally with alternating single and double bonds (because there is no `-xA` flag):

```shell
$ obabel -:"c1ccccc1" -ocml
<?xml version="1.0"?>
<molecule xmlns="http://www.xml-cml.org/schema">
 <atomArray>
  <atom id="a1" elementType="C" hydrogenCount="1"/>
  <atom id="a2" elementType="C" hydrogenCount="1"/>
  <atom id="a3" elementType="C" hydrogenCount="1"/>
  <atom id="a4" elementType="C" hydrogenCount="1"/>
  <atom id="a5" elementType="C" hydrogenCount="1"/>
  <atom id="a6" elementType="C" hydrogenCount="1"/>
 </atomArray>
 <bondArray>
  <bond atomRefs2="a1 a2" order="1"/>
  <bond atomRefs2="a2 a3" order="2"/>
  <bond atomRefs2="a3 a4" order="1"/>
  <bond atomRefs2="a4 a5" order="2"/>
  <bond atomRefs2="a5 a6" order="1"/>
  <bond atomRefs2="a1 a6" order="2"/>
 </bondArray>
</molecule>
1 molecule converted
```

or with an explicit bond order "aromatic" while `-xA` is used:

```shell
$ obabel -:"c1ccccc1" -ocml -xA
<?xml version="1.0"?>
<molecule xmlns="http://www.xml-cml.org/schema">
 <atomArray>
  <atom id="a1" elementType="C" hydrogenCount="1"/>
  <atom id="a2" elementType="C" hydrogenCount="1"/>
  <atom id="a3" elementType="C" hydrogenCount="1"/>
  <atom id="a4" elementType="C" hydrogenCount="1"/>
  <atom id="a5" elementType="C" hydrogenCount="1"/>
  <atom id="a6" elementType="C" hydrogenCount="1"/>
 </atomArray>
 <bondArray>
  <bond atomRefs2="a1 a2" order="A"/>
  <bond atomRefs2="a2 a3" order="A"/>
  <bond atomRefs2="a3 a4" order="A"/>
  <bond atomRefs2="a4 a5" order="A"/>
  <bond atomRefs2="a5 a6" order="A"/>
  <bond atomRefs2="a1 a6" order="A"/>
 </bondArray>
</molecule>
1 molecule converted
```

Is the OpenBabel package you use of similar age of compilation than the one I use and with similar/the same bug fixes considered?  That is (as provided by Debian 13/trixie) for me:

```shell
$ obabel -v
No input file or format spec or possibly a misplaced option.
Most options must come after the input files. (-i -o -O -m can be anywhwere.)

Open Babel 3.1.1 -- Jan  4 2023 -- 09:58:24
Usage:
obabel [-i<input-type>] <infilename> [-o<output-type>] -O<outfilename> [Options]
Try  -H option for more information.
```

**Comment by kmlefran** [comment]
Moving forward I'll be discussing my Mac installation of OpenBabel, which I just updated to be sure

```
% obabel -v
No input file or format spec or possibly a misplaced option.
Most options must come after the input files. (-i -o -O -m can be anywhwere.)

Open Babel 3.1.1 -- Nov 20 2023 -- 13:12:12
Usage:
obabel [-i<input-type>] <infilename> [-o<output-type>] -O<outfilename> [Options]
Try  -H option for more information.
```

Running the below on the -xA option cml has the issue and write radicals as described

```
% obabel aromatic.cml -ocan 
[CH]1[CH][CH][CH][CH][CH]1
```

**Comment by nbehrnd** [comment]
I speculate there could be a bug because of the following two different outcomes.

0) generation of the aromatic.cml (bond orders marked by A)

```shell
$ obabel -:"c1ccccc1" -ocml --gen3d -xA > ~/Desktop/test.cml
1 molecule converted
$ cat ~/Desktop/test.cml
$ cat ~/Desktop/test.cml 
<?xml version="1.0"?>
<molecule xmlns="http://www.xml-cml.org/schema">
 <atomArray>
  <atom id="a1" elementType="C" hydrogenCount="1" x3="1.382807" y3="-0.221802" z3="0.005621"/>
  <atom id="a2" elementType="C" hydrogenCount="1" x3="0.506827" y3="-1.306766" z3="-0.007235"/>
  <atom id="a3" elementType="C" hydrogenCount="1" x3="-0.871185" y3="-1.090814" z3="-0.015111"/>
  <atom id="a4" elementType="C" hydrogenCount="1" x3="-1.372721" y3="0.210822" z3="-0.004526"/>
  <atom id="a5" elementType="C" hydrogenCount="1" x3="-0.496795" y3="1.296205" z3="0.011557"/>
  <atom id="a6" elementType="C" hydrogenCount="1" x3="0.881190" y3="1.080097" z3="0.012965"/>
  <atom id="a7" elementType="H" hydrogenCount="0" x3="2.456430" y3="-0.390211" z3="0.009514"/>
  <atom id="a8" elementType="H" hydrogenCount="0" x3="0.897545" y3="-2.320834" z3="-0.010961"/>
  <atom id="a9" elementType="H" hydrogenCount="0" x3="-1.553723" y3="-1.935967" z3="-0.029527"/>
  <atom id="a10" elementType="H" hydrogenCount="0" x3="-2.446585" y3="0.379351" z3="-0.008513"/>
  <atom id="a11" elementType="H" hydrogenCount="0" x3="-0.887676" y3="2.310028" z3="0.022209"/>
  <atom id="a12" elementType="H" hydrogenCount="0" x3="1.563844" y3="1.925102" z3="0.020756"/>
 </atomArray>
 <bondArray>
  <bond atomRefs2="a1 a2" order="A"/>
  <bond atomRefs2="a2 a3" order="A"/>
  <bond atomRefs2="a3 a4" order="A"/>
  <bond atomRefs2="a4 a5" order="A"/>
  <bond atomRefs2="a5 a6" order="A"/>
  <bond atomRefs2="a1 a6" order="A"/>
  <bond atomRefs2="a1 a7" order="1"/>
  <bond atomRefs2="a2 a8" order="1"/>
  <bond atomRefs2="a3 a9" order="1"/>
  <bond atomRefs2="a4 a10" order="1"/>
  <bond atomRefs2="a5 a11" order="1"/>
  <bond atomRefs2="a6 a12" order="1"/>
 </bondArray>
</molecule>
```

option a) read by an old instance of Avogadro 1.97 which displays a flat benzene, all bonds single bonds.  If I export this as a .mol file, and request the conversion into a SMILES string, the result is upper case cyclohexane `C1CCCCC1`.

option b) I read `aromatic.cml` by Jmol (version 16.1.41) which display the formally alternating single-double bond pattern, the export to .mol and subsequent request (to openbabel) to assign a SMILES string is lower case benzene, `c1cccccc1`.

Based on this observation, perhaps openbabel indeed has a difficulty to process the .cml file into other file formats.  Does your workflow require the intermediate .cml format?  As it (contrasting to .smi) retains 3D coordinates, were the use e.g., .mol/.sdf an alternative good enough until one of maintainers can check the issue?

**Comment by kmlefran** [comment]
I have a large collection of .cml files that I am using, with bond orders labelled as aromatic rather than in Kekule form, so using .cml in this process is mandatory.

I have some potential workarounds, like converting to .mol, then changing the '5' bond orders in the bond block to '4's, then going from there, or maybe just going back to Open Babel 2.4.1 for now. I'll use one of these until this gets looked at

**Comment by fredrikw** [comment]
The title of this should rather be "CML Aromatic Bonds not being read"... I have prepared a fix in #2649
